### PR TITLE
Add support for extensions and extension schemas

### PR DESCRIPTION
**Pull request by heimweh**
This PR adds support for managing extensions and extension schemas.

### CODE REVIEW

File: pagerduty/extension.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import "fmt"
+
+// ExtensionService handles the communication with extension related methods
+// of the PagerDuty API.
+type ExtensionService service
+
+// Extension represents an extension.
+type Extension struct {
+	Extension        *Extension                `json:"extension,omitempty"`
```
Comment (pdecat) [inline pagerduty/extension.go:11]: Why is there a pointer to the struct being defined? Can an extension link to another extension?

I've noticed this was done on most struct definitions in `go-pagerduty` but I don't see how this matches with the API.

File: pagerduty/extension_schema.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import "fmt"
+
+// ExtensionSchemaService handles the communication with extension schemas related methods
+// of the PagerDuty API.
+type ExtensionSchemaService service
+
+// ExtensionSchema represents an extension schema.
+type ExtensionSchema struct {
+	ExtensionSchema *ExtensionSchema `json:"extension_schema,omitempty"`
```
Comment (pdecat) [inline pagerduty/extension_schema.go:11]: Same as above, why is there a pointer to the struct being defined? Can an extension_schema link to another extension_schema?

**Comment by nelg** [comment]
+1 for this feature

File: pagerduty/extension.go (hunk 1 above)
Comment (heimweh) [reply pagerduty/extension.go:11]: Hi @pdecat, this is merely a work-around used internally within the client to marshal/unmarshal requests/responses, which I'm planning to replace with proper request/response structs in the near future.

**Pull request merged**